### PR TITLE
PD-2445 Add dependency check

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Adds a dependency check to CI steps. Fixes issues found in the check

🎟 _Jira ticket:_ [PD-2445](https://jira.mongodb.org/browse/PD-2445)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/text-area/package.json [hunk 1]
```diff
@@ -20,10 +20,11 @@
     "access": "public"
   },
   "dependencies": {
-    "@leafygreen-ui/lib": "^9.4.2",
     "@leafygreen-ui/emotion": "^4.0.1",
     "@leafygreen-ui/hooks": "^7.3.1",
-    "@leafygreen-ui/interaction-ring": "^4.0.1",
+    "@leafygreen-ui/icon": "^11.10.2",
+    "@leafygreen-ui/leafygreen-provider": "^2.3.3",
```
Comment (shaneeza) [inline packages/text-area/package.json:26]: Should this be in `peerDependencies`?

File: packages/tooltip/package.json [hunk 2]
```diff
@@ -24,11 +24,12 @@
   },
   "dependencies": {
     "@leafygreen-ui/emotion": "^4.0.1",
+    "@leafygreen-ui/hooks": "^7.3.1",
     "@leafygreen-ui/icon": "^11.10.2",
-    "@leafygreen-ui/popover": "^9.0.1",
+    "@leafygreen-ui/leafygreen-provider": "^2.3.3",
```
Comment (shaneeza) [inline packages/tooltip/package.json:29]: same here

**Review by shaneeza — COMMENTED** [review]
Should `tsconfig` for each file also get updates?

**Comment by TheSonOfThomp** [comment]
> Should `tsconfig` for each file also get updates?

@shaneeza Added a `--fix-tsconfig` flag, and ran it 👍

**Review by shaneeza — APPROVED** [review]
Looks good, its pretty cool that you could do all of this with scripts!

**Pull request merged**